content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,243 @@
+field *must* be set on any of the following options: :guilabel:`Service`, :guilabel:`Event Ticket`,
+:guilabel:`Event Booth`, or :guilabel:`Course`.
+
+.. image:: milestone/product-type-field.png
+   :align: center
+   :alt: The invoicing policy field drop-down menu with options on product form.
+
+With any of those :guilabel:`Product Type` options selected, choose :guilabel:`Based on Milestones`
+from the :guilabel:`Invoicing Policy` drop-down menu.
+
+.. image:: milestone/invoicing-policy-field.png
+   :align: center
+   :alt: The invoicing policy field drop-down menu with options on product form.
+
+Beneath that is the :guilabel:`Create on Order` field.
+
+.. note::
+    Leaving it on the default :guilabel:`Nothing` option wont negatively affect the desired
+    workflow. However, a project *must* then be created directly from a sales order form with that
+    specific product. Once a project is created *then* milestones and tasks can be created and
+    configured.
+
+To ensure workflows are as seamless as possible, it is recommended that an option in the
+:guilabel:`Create on Order` field is selected.
+
+When the :guilabel:`Create on Order` default option of :guilabel:`Nothing` is clicked, a drop-down
+menu is revealed with the following options:
+
+- :guilabel:`Task`: Odoo creates a task related to this milestone product in the *Projects* app when
+  this specific product is ordered.
+- :guilabel:`Project & Task`: Odoo creates a project and task related to this milestone product in
+  the *Projects* app when this specific product is ordered.
+- :guilabel:`Project`: Odoo creates a project related to this milestone product in the *Projects*
+  app when this specific product is ordered.
+
+When :guilabel:`Task` is selected, a :guilabel:`Project` field appears. In this field, select to
+which existing project in the *Projects* app this created task should be connected.
+
+.. image:: milestone/task-option-project-field.png
+   :align: center
+   :alt: The Project field appears when the Task option is selected in Create on Order field.
+
+When :guilabel:`Project & Task` or :guilabel:`Project` is selected, two new fields appear:
+:guilabel:`Project Template` and :guilabel:`Workspace Template`.
 
-You have to set the product type as *Service* under general
-information and select *Milestones* in the sales tab.
+.. image:: milestone/project-task-option-project-workspace-fields.png
+   :align: center
+   :alt: The Project template and workspace template fields that appear on milestone product.
 
-.. image:: milestone/milestone01.png
-    :align: center
+The :guilabel:`Project Template` field provides template options to use for the project that will be
+created when this specific product is ordered.
+
+The :guilabel:`Workspace Template` field provides template options to use for the workspace that
+will be automatically generated for the project when this specific product is ordered.
```

Comment:
I would specify that the Workspace template is for the Documents app and not the Projects app.